ckanext/s3filestore/uploader.py

```diff
@@ -159,21 +167,25 @@ def get_signed_url_to_key(self, key_path):
         client = self.get_s3_client()
 
         # check whether the object exists in S3
-        client.head_object(Bucket=self.bucket_name, Key=key_path)
+        metadata = client.head_object(Bucket=self.bucket_name, Key=key)
 
         if self.signed_url_cache_enabled:
             redis_conn = connect_to_redis()
-            cache_key = _get_cache_key(key_path)
+            cache_key = _get_cache_key(key)
             cache_url = redis_conn.get(cache_key)
             if cache_url:
-                log.debug('Returning cached URL for path %s', key_path)
+                log.debug('Returning cached URL for path %s', key)
                 return cache_url
             else:
-                log.debug('No cache found for %s; generating a new URL', key_path)
+                log.debug('No cache found for %s; generating a new URL', key)
 
+        params = {'Bucket': self.bucket_name,
+                  'Key': key}
+        if metadata['ContentType'] != 'application/pdf':
+            params['ResponseContentDisposition'] = 'attachment; filename=' + key
```

Comment:
should this be controlled by a flag to enable or disable  ContentDisposition/Download only?

Reply:
Since it's a security matter, we probably shouldn't have a flag to disable it, no. But perhaps there could be a configurable whitelist of content types that can be displayed inline, defaulting to PDF?

Reply:
yep, make it a config option, for a coma delimited list, allow '*' as an option to allow all to bypass ContentDisposition so that others can continue to have the insecure way ckan is.